lib/feedback.js

```diff
@@ -169,60 +150,125 @@ function feedback(system) {
 				}
 			}
 		}
+
+		self.system.emit('schedule_get_all_feedbacks', (scheduler_feedbacks) => {
+			for (const feedback of scheduler_feedbacks) {
+				if (feedback.instance_id == instance_id) {
+					fbs.push(feedback)
+				}
+			}
+		})
+
 		cb(fbs)
 	})
 
-	system.on('feedback_instance_check', function (instance, type) {
-		//debug('Instance ' + instance.label + ' wants us to check banks (' + type + ')');
-		for (var page in self.feedbacks) {
-			for (var bank in self.feedbacks[page]) {
-				// Iterate through feedbacks on this bank
-				if (self.feedbacks[page][bank] !== undefined) {
-					for (var i in self.feedbacks[page][bank]) {
-						var feedback = self.feedbacks[page][bank][i]
+	system.on('feedback_definition_get', function (instance_id, type, cb) {
+		if (
+			self.feedback_definitions[instance_id] !== undefined &&
+			self.feedback_definitions[instance_id][type] !== undefined
+		) {
+			cb(self.feedback_definitions[instance_id][type])
+		} else {
+			cb(undefined)
+		}
+	})
 
-						if (type !== undefined && feedback.type != type) {
-							continue
-						}
+	/** Check if a feedback is supposed to be checked, for some constraints */
+	function feedback_check_constraints(constraint, feedback) {
```

Comment:
i'd prefer to just make this a class function now, since that's what I'm going to do when merging into feat/registry

Reply:
I dont really like how this separates the definition from the usage, but ive done it